ql/experimental/credit/syntheticcdo.cpp

```diff
@@ -107,11 +107,12 @@ namespace QuantLib {
     Real SyntheticCDO::protectionLegNPV() const {
         calculate();
         if(side_ == Protection::Buyer) return -protectionValue_;
-        return premiumValue_;
+        return protectionValue_;
     }
 
     Rate SyntheticCDO::fairPremium () const {
         calculate();
+        QL_REQUIRE(premiumValue_ == 0, "Attempted divide by zero while calculating syntheticCDO premium.");
```

Comment:
Except we must require that the value is `premiumValue_ != 0`.

On the bright side, with the new commit the checks realized that you signed the CLA.

Reply:
Sorry about that, it's the danger when you're pushing stuff last thing at night! Have updated.